_test/check-exercises.sh

```diff
@@ -27,14 +34,33 @@ for exercise in exercises/*/tests; do
         done
 
         # Run the test and get the status
-        cargo test
+        if [ -n "$DENYWARNINGS" ]; then
+            # In DENYWARNINGS mode, unset -e so that we run all tests.
+            # This allows us to see all warnings.
+            # No-run mode so we see no test output.
+            # Quiet mode so we see no compile output
+            # (such as "Compiling"/"Downloading").
+            # Compiler errors will still be shown though.
+            # Both flags are necessary to keep things quiet.
+            set +e
+            cargo test --quiet --no-run
+            status=$?
+            set -e
+            return $status
```

Comment:
NOPE, bash doesn't let you `return` out of a subshell (but `sh` does). Fine, I have other ideas.